ddht/v5_1/client.py

```diff
@@ -437,3 +463,126 @@ async def topic_query(
+                    request.receiver_node_id,
+                )
+
+                async with subscription_ctx as subscription:
+
+                    async for response in subscription:
+                        if response.message.total == 0:
+                            raise ValidationError(
+                                "Invalid `total` counter in response: "
+                                f"total={response.message.total}"
+                            )
+
+                        if response.request_id != request_id:
+                            continue
+                        else:
+                            if response_counter == 0:
+                                head_response_total = response.message.total
+
+                            if head_response_total != response.message.total:
+                                raise ValidationError(
+                                    "Inconsistent message total. Received a FoundNodesMessage with "
+                                    f"a total of {response.message.total}, "
+                                    f"expected a total of {head_response_total}"
+                                )
+
+                            validate_found_nodes_distances(
+                                response.message.enrs,
+                                request.receiver_node_id,
+                                request.message.distances,
+                            )
+
+                            response_counter += 1
+                            try:
+                                await send_channel.send(response)
+                            except (trio.BrokenResourceError, trio.ClosedResourceError):
+                                break
+
+                        if response_counter == head_response_total:
+                            all_responses_received.set()
+
+        if scope.cancelled_caught:
+            client.logger.debug(
+                "Stream find nodes request disconnected: request=%s message_type=%s",
+                request,
+                request_id,
+            )
+
+    async def _monitor_done(
+        all_responses_received: trio.Event,
+        send_channel: trio.abc.SendChannel[InboundMessage[FoundNodesMessage]],
+    ) -> None:
+        async with send_channel:
+            while True:
+                await all_responses_received.wait()
+                break
+
+    all_responses_received = trio.Event()
+
+    async with trio.open_nursery() as nursery:
+        with client.request_tracker.reserve_request_id(
+            node_id, request_id
+        ) as request_id:
+            request = AnyOutboundMessage(
+                FindNodeMessage(request_id, tuple(distances)), endpoint, node_id,
+            )
+
+            send_channel, receive_channel = trio.open_memory_channel[
+                InboundMessage[FoundNodesMessage]
+            ](4)
+            nursery.start_soon(
+                # mypy expects an Awaitable, interprets this as a Coroutine
```

Comment:
mypy interprets `_stream_find_nodes_response` as a `Coroutine` but expects an `Awaitable` - but from what I understand, `Coroutine` is a subclass of `Awaitable`. Any thoughts as to what could be problematic here?

Reply:
So I think what you're actually seeing is that `start_soon` doesn't like functions with more than 4 arguments.

```python
async def foo() -> None:
    async def arg_4(a: int, b: int, c: int, d: int) -> None:
        return
    async def arg_5(a: int, b: int, c: int, d: int, e: int) -> None:
        return

    async with trio.open_nursery() as nursery:
        nursery.start_soon(arg_4, 1, 2, 3, 4)
        nursery.start_soon(arg_5, 1, 2, 3, 4, 5)  # mypy gets mad about this.
```

You can work around this by simply using a `type: ignore`